src/cryptoadvance/specter/notifications/websockets_server_client.py

```diff
@@ -0,0 +1,332 @@
+                                                            ┌───────────────────────┐
+                                                            │                       │
+                                                            │  NotificationManager  │
+                                                            │                       │
+                                                            └───────────────────────┘
+    2.    A UI_Notification creates a message for the browser to show
+        ┌───────────────────────┐                           ┌───────────────────────┐
+        │                       │ websockets_client.send    │                       │
+        │ JSConsoleNotifications├─────────────────────────► │   WebsocketsClient    │
+        │                       │                           │                       │
+        └───────────────────────┘                           └───────────┬───────────┘
+                                                                        │
+                                                                        │
+                                                                        │ websockets_client.send
+                                                                        │
+                                                                        ▼
+        ┌───────────────────────┐                           ┌───────────────────────┐
+        │                       │                           │                       │
+        │ Browser javascript    │     websocket.send        │                       │
+        │                       │◄──────────────────────────┤  WebsocketsServer     │
+        │ websocket.on_message  │                           │                       │
+        │                       │                           │                       │
+        └───────────────────────┘                           └───────────────────────┘
+    """
+
+    def __init__(self, notification_manager, user_manager):
+        logger.info(f"Create {self.__class__.__name__}")
+
+        self.broadcaster_tokens = list()
+        self.connections = list()
+        self.notification_manager = notification_manager
+        self.user_manager = user_manager
+
+    def get_broadcaster_tokens(self):
+        return [d["user_token"] for d in self.broadcaster_tokens]
+
+    def get_token_of_websocket(self, websocket):
+        for d in self.connections:
+            if d["websocket"] == websocket:
+                return d["user_token"]
+
+    def get_connections_by_token(self, user_token):
+        connections = []
+        for d in self.connections:
+            if d["user_token"] == user_token:
+                connections.append(d["websocket"])
+        return connections
+
+    def get_user_of_user_token(self, user_token):
+        for u in self.user_manager.users:
+            if u.websocket_token == user_token:
+                return u.username
```

Comment:
Can we explicitely return something (or throw an exception) if the user is not found?